### PR TITLE
Whitelist Access Control and Commands for AI Config

### PR DESCRIPTION
**Pull request by Seitk**
## Changes

- Added whitelist on phone numbers
- Restructured config with extendable commands interface
- Added config for transcription and tts

### Sample of commands:

```
Available commands:
	!config dalle size <value> - Set dalle size to <value>
	!config chat id - Get the ID of the chat
	!config general whitelist <value> - Set whitelisted phone numbers
	!config gpt maxModelTokens <value> - Set max model tokens value
	!config transcription enabled <value> - Toggle if transcription is enabled
	!config transcription mode <value> - Set transcription mode
	!config tts enabled <value> - Toggle if TTS is enabled

Available values:
	dalle size: 256x256, 512x512, 1024x1024
	gpt maxModelTokens: integer
	transcription enabled: true, false
	transcription mode: local, speech-api, whisper-api, openai
	tts enabled: true, false
```

## To Add Whitelist

- run `!config chat id` to get the user phone number id (like 85299992222@c.us)
- paste the id into whitelist command: `!config general whitelist 85299992222@c.us`

Remarks:
The whitelist stored in memory, persist the value into environment variable if needed

### CODE REVIEW

**Review by LuckySpec — CHANGES_REQUESTED** [review]
good work

**Comment by verfasor** [comment]
Same here. My system reproduced the same error as @LuckySpec shared.

File: .env-example [hunk 1]
```diff
@@ -34,6 +37,9 @@ AI_CONFIG_PREFIX=!config
 PROMPT_MODERATION_ENABLED = true
 PROMPT_MODERATION_BLACKLISTED_CATEGORIES = ["hate", "hate/threatening", "self-harm", "sexual", "sexual/minors", "violence", "violence/graphic"]
 
+# Access control, only allow whatsapp-chatgpt to react to specific phone numbers, comma-separated
+WHITELISTED_PHONE_NUMBERS=85233332222@c.us,85299996666@c.us,85288882222-12345678@g.us
```
Comment (navopw) [inline .env-example:41]: Remove the numbers from examples, put them in #

Comment (LuckySpec) [reply .env-example:41]: the same, when I put the number in the env everything works, but when a number that is not in the whitelist send a command the error occurs

**Comment by Seitk** [comment]
let me take a look

**Comment by Seitk** [comment]
Somehow a commit is missed when I do rebase with stash, fixed

File: .env-example (hunk 1 above)
Comment (navopw) [reply .env-example]: You should not put numbers into the example, leave it empty and people have to put their own numbers in there.

Also make a boolean to enable/disable the whitelisting

File: src/commands/transcription.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+const enabled: ICommandDefinition = {
+	help: "<value> - Toggle if transcription is enabled",
+	hint: "true, false",
+	data: config.transcriptionEnabled,
+	execute: function (message: Message, valueStr?: string) {
+		if (["true", "false"].indexOf(valueStr || "") < 0) {
+			message.reply(`Invalid value, please specify true or false`);
+			return;
+		}
+		this.data = valueStr == "true";
```
Comment (navopw) [inline src/commands/transcription.ts:25]: transcriptionEnabled is a boolean. The commands should change the value in the config, then you also dont have to use `getConfig("tts", "enabled")`

Comment (Seitk) [reply src/commands/transcription.ts:25]: actually I think we could consider to move all config into commands instead, so every config is overridable in runtime

Comment (navopw) [reply src/commands/transcription.ts:25]: Ok fine, I will review & test this in the next days and merge it. (Quite busy right now)

Thank you for this contribution.

**Review by navopw — APPROVED** [review]

**Pull request merged**